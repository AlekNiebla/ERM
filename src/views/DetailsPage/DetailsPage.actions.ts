import { StoreDispatch, ThunkAction } from "Types";
import movieService, { resultType } from "Services/movieService";
import globalActionTypes, { toggleFetchStatus } from "Root/global.actions";

const enum detailsActionTypes {
  getMovieDetails = "GET_MOVIE_DETAILS_QUERY",
  getMovieDetailsResponse = "GET_MOVIE_DETAILS_RESPONSE"
}

export default detailsActionTypes;

type getMovieByIDType = (
  id: string,
  dispatch: StoreDispatch<Error | resultType | boolean>
) => void;

const getMovieByID: getMovieByIDType = async (id, dispatch) => {
  await movieService
    .getMovieByID(id)
    .then((result: resultType) => {
      dispatch({
        type: detailsActionTypes.getMovieDetailsResponse,
        payload: result
      });
    })
    .catch((error: Error) => {
      dispatch({
        type: globalActionTypes.fetchError,
        payload: error
      });
    });
};

export const fetchMovieById: ThunkAction<string, boolean> = id => dispatch => {
  getMovieByID(id, dispatch);
  dispatch(toggleFetchStatus(true));

  return {
    type: detailsActionTypes.getMovieDetails
  };
};
