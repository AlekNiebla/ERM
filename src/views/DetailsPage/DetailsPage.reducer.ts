import { IMovie, GenericAction } from "Types";
import { IMovieDetails } from "./components/MovieDetails/MovieDetails";

import detailsActionTypes from "./DetailsPage.actions";

interface DetailsPageReducerState {
  details: false | IMovieDetails;
  similarMovies: false | IMovie[];
}

type detailsPageReducerType = (
  state: DetailsPageReducerState,
  action: GenericAction<any>
) => DetailsPageReducerState;

const initialState: DetailsPageReducerState = {
  details: false,
  similarMovies: false
};

const detailsPageReducer: detailsPageReducerType = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case detailsActionTypes.getMovieDetailsResponse:
      return {
        ...state,
        details: action.payload.movie,
        similarMovies: action.payload.similar
      };
    case detailsActionTypes.getMovieDetails:
    default:
      return state;
  }
};

export default detailsPageReducer;
