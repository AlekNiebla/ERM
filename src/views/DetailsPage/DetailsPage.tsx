import * as React from "react";
// components
import Header from "./../../components/Header/Header";
import MovieDetails from "./components/MovieDetails/MovieDetails";
import MovieGrid from "./../../components/MovieGrid/MovieGrid";
import Footer from "./../../components/Footer/Footer";
import Button from "./../../components/Button/Button";

const DetailsPage = () => {
  const SearchLinkButton = <Button type="primary" label="search" />;
  return (
    <div>
      <Header actionItem={SearchLinkButton}>
        <MovieDetails />
      </Header>
      <div>
        <MovieGrid />
      </div>
      <Footer />
    </div>
  );
};
export default DetailsPage;
