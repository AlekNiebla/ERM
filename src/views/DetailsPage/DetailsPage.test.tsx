import * as React from "react";
import { shallow } from "enzyme";
// components
import DetailsPage from "./DetailsPage";

describe("DetailsPage component...", () => {
  it("... renders without errors", () => {
    const wrapper = shallow(<DetailsPage />);
    expect(wrapper.find(DetailsPage)).toBeTruthy();
  });
  it("... matches snapshot", () => {
    const wrapper = shallow(<DetailsPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
