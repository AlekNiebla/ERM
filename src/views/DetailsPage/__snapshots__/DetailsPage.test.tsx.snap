// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component... ... matches snapshot 1`] = `ShallowWrapper {}`;
